Library/ViewModels/ProjectPamphletViewModel.swift

```diff
@@ -80,13 +86,36 @@ public final class ProjectPamphletViewModel: ProjectPamphletViewModelType, Proje
     let freshProjectAndRefTag = freshProjectAndRefTagEvent.values()
 
     let ctaButtonTapped = freshProjectAndRefTag
-      .takeWhen(self.backThisProjectTappedProperty.signal)
-      .map { project, refTag in
-        (project, refTag)
+      .takePairWhen(self.pledgeCTAButtonTappedProperty.signal)
+      .map(unpack)
+
+    let goToManagePledge = ctaButtonTapped
+      .filter(canShowManageViewPledgeScreen(_:_:state:))
+      .map { (project, refTag, _) -> PledgeData in
+        let r = reward(from: project.personalization.backing, inProject: project)
```

Comment:
Could we name this `reward` (it should fit on the screen and be more descriptive)?

Reply:
I tried, but keep getting the error `Variable used within its own initial value`

Reply:
Right, good point!

How about something like this:

```swift
.map { (project, refTag, _) -> PledgeData in
        PledgeData(
          project: project,
          reward: reward(from: project.personalization.backing, inProject: project),
          refTag: refTag
        )
      }
```

or

```swift
.map {
        PledgeData(
          project: $0.0, reward: reward(from: $0.0.personalization.backing, inProject: $0.0), refTag: $0.1
        )
      }
```
(but that might be more than confusing due to the repetition of those variables)

Reply:
Since this is a language limitation I'd be OK with leaving it as is .. your decision! 🥂